src/spec-configuration/containerFeaturesOCI.ts

```diff
@@ -0,0 +1,197 @@
+                'org.opencontainers.image.title': string;
+            };
+        }
+    ];
+}
+
+export function getOCIFeatureSet(output: Log, identifier: string, options: boolean | string | Record<string, boolean | string | undefined>, manifest: OCIManifest): FeatureSet {
+
+    const featureRef = getFeatureRef(output, identifier);
+
+    const feat = {
+        id: featureRef.id,
+        name: featureRef.id,
+        included: true,
+        value: options
+    };
+
+    let featureSet: FeatureSet = {
+        sourceInformation: {
+            type: 'oci',
+            manifest: manifest,
+            featureRef: featureRef,
+
+        },
+        features: [feat],
+    };
+
+    return featureSet;
+}
+
+export function getFeatureRef(output: Log, identifier: string): OCIFeatureRef {
+
+    // ex: ghcr.io/codspace/features/ruby:1
+    const splitOnColon = identifier.split(':');
+    const id = splitOnColon[0];
+    const version = splitOnColon[1] ? splitOnColon[1] : '1';
+
+    const splitOnSlash = id.split('/');
+    const featureName = splitOnSlash[splitOnSlash.length - 1];
+    const owner = splitOnSlash[1];
+    const registry = splitOnSlash[0];
+    const namespace = splitOnSlash.slice(1, -1).join('/');
+
+    output.write(`identifier: ${identifier}`);
+    output.write(`id: ${id}`);
+    output.write(`version: ${version}`);
+    output.write(`featureName: ${featureName}`);
+    output.write(`owner: ${owner}`);
+    output.write(`namespace: ${namespace}`);
+    output.write(`registry: ${registry}`);
+
+    return {
+        id,
+        version,
+        featureName,
+        owner,
+        namespace,
+        registry
+    };
+}
+
+// Validate if a manifest exists and is reachable about the declared feature.
+export async function validateOCIFeature(output: Log, env: NodeJS.ProcessEnv, identifier: string): Promise<OCIManifest | undefined> {
+    const featureRef = getFeatureRef(output, identifier);
+
+    const manifestUrl = `https://${featureRef.registry}/v2/${featureRef.namespace}/${featureRef.featureName}/manifests/${featureRef.version}`;
+    output.write(`manifest url: ${manifestUrl}`, LogLevel.Trace);
+    const manifest = await getFeatureManifest(output, env, manifestUrl);
+
+    return manifest;
+}
+
+// Download a feature from which a manifest was previously downloaded.
+export async function fetchOCIFeature(output: Log, env: NodeJS.ProcessEnv, featureSet: FeatureSet, ociCacheDir: string, featCachePath: string): Promise<boolean> {
+
+    if (featureSet.sourceInformation.type !== 'oci') {
+        output.write(`FeatureSet is not an OCI featureSet.`);
+        throw new Error('FeatureSet is not an OCI featureSet.');
+    }
+
+    const blobUrl = `https://${featureSet.sourceInformation.featureRef.registry}/v2/${featureSet.sourceInformation.featureRef.namespace}/${featureSet.sourceInformation.featureRef.featureName}/blobs/${featureSet.sourceInformation.manifest?.layers[0].digest}`;
```

Comment:
Ensuring the feature declares media type will also allow us to be confident that `featureSet.sourceInformation.manifest?.layers[0].digest` and any other fields of fields exist.

Reply:
Added.